### PR TITLE
Remove nogroup group

### PR DESCRIPTION
**Pull request by janisz**
Fixes: https://jira.mesosphere.com/browse/COPS-5220
Refs: 
* https://github.com/dcos/dcos/pull/6166
* https://github.com/dcos/dcos-launch/pull/244

### CODE REVIEW

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3 @fatz  Can you review, please?

**Review by fatz — CHANGES_REQUESTED** [review]
this drops immediately support for previous DC/OS versions and will make clusters running previous versions unmaintainable. 

This needs version matching.

- Which releases will support the absence of nogroup?
- What is the migration path? Was it tested droping the group on upgrade?

**Comment by janisz** [comment]
@drewvanstone Can you answer above?

**Comment by drewvanstone** [comment]
This change was made in 1.14 (2.0) release, but will be back ported back to 1.12 release.

We've been attempting to test this out on soak to be sure everything works properly. But it seems soak still has the nogroup user. I'm not sure how deployments there work, but if they are fresh installs, then we're hoping this PR will help ensure the group is not created. Once we've run in soak for a few days without nogroup, then we'll finish back porting.

Open to any suggestions though.

**Comment by janisz** [comment]
I tried to find who is using `nogroup` and there is no evidence of using it in dcos/dcos repo 
```bash
git grep nogroup $(git rev-list --all) -- './*' \
    ':!*.md' ':!gen/build_deploy/bash.py' \
    ':!packages/adminrouter/extra/src/test-harness/modules/mocker/endpoints/generic.py' \
    ':!dcos_installer/action_lib.py' \
    ':!cloud_images/lib/el7/configure_dcos_system.sh' \
    ':!cloud_images/centos7/install_prereqs.sh' \
    ':!gen/installer/bash.py' \
    ':!ext/dcos-installer/dcos_installer/action_lib.py' \
    ':!ext/dcos-installer/dcos_installer/action_lib/__init__.py' \
    ':!test_util/test_installer_ccm.py'
```

**Comment by fatz** [comment]
@drewvanstone whats your understanding of backporting? Does this mean replacing every release from 1.12 on? 

If backporting is meant just adding a new patch release to 1.12 and 1.13 previous patches would still need this user.

So which *exact* versions will support the absence of nogroup?

> What is the migration path?
- How do we handle running clusters? 
- Should the user be dropped or ignored?

As the name of this module is prerequisites it will run before any DC/OS update is done which means in case of dropping the group it could potentially cause some problems.

What about file permissions? Is this group used in combination with group permissions on files? That could also lead to issues while upgrading

**Comment by drewvanstone** [comment]
For now, the only version that matters is 2.0. We need to test everything together on a deployed cluster, and then we can work out the exact versions for any sort of backport. 

For running clusters going through an upgrade, the group should be dropped. It doesn't look like this group has been used in a long time. It was needed by nginx long ago, but was replaced by the `adminrouter` group.

Hopefully that answers your question. It's quite difficult to test these changes all together on a new cluster, so we're relying on the standard deployment process to soak to find any issues.

**Comment by fatz** [comment]
But that's what I'm saying. Once this is merged to master its live for any Universal Installer 0.2 based cluster.

So we have to be very careful with this. 

So first of it all this needs a version condition... meaning everything greater than 2.0 might be fine.

Still unclear is the case of an upgrade. The safest way would be ignore rather than deleting the group. So new nodes `(>= 2.0.0 )` will be fine anyways and upgrade from previous versions means just ignoring the group.

**Comment by fatz** [comment]
version checking could be done like this https://github.com/dcos/dcos-ansible/blob/master/roles/dcos_agent/tasks/dcos_upgrade.yml#L23

**Comment by makkes** [comment]
Aren't we just ignoring the group when we apply this patch?

**Comment by fatz** [comment]
> For running clusters going through an upgrade, the group should be dropped. It doesn't look like this group has been used in a long time. It was needed by nginx long ago, but was replaced by the adminrouter group.

currently yes but it had be clarified. 

So for now my clear suggestion is having a version check on >= 2.0 and if it doesn't match we still create that group but we also don't delete it. Which is to me the safest way supporting this

**Comment by janisz** [comment]
@fatz @drewvanstone Can you take a look now?

File: roles/dcos_requirements/tasks/main.yml
Comment (fatz) [inline roles/dcos_requirements/tasks/main.yml]: looking much better but IMO you should use dcos['version'] instead as this variable is not yet set. 

Waiting for the CI but expecting an error

Comment (janisz) [reply roles/dcos_requirements/tasks/main.yml]: Fixed

File: roles/dcos_requirements/tasks/main.yml
Comment (makkes) [inline roles/dcos_requirements/tasks/main.yml]:
```suggestion
- name: "Ensure 'nogroup' group exists for DC/OS version < 2.0"
```

**Review by makkes — CHANGES_REQUESTED** [review]

**Comment by janisz** [comment]
@fatz Is this change tested in CI? I cannot find a log for it

**Review by drewvanstone — APPROVED** [review]
@fatz looks good to me. What do you think?

**Review by makkes — APPROVED** [review]

**Review by fatz — APPROVED** [review]
lgtm

**Pull request merged**
